Update read_mri_data_to_function

cdaversin opened the pull request:
Updates to read_mri_data_to_function :

-  Can read nifti Mri data (.nii / .nii.gz), in addition to mgh images (.mgz)
-  Checks the orientation of the given data (has to be RAS)
-  Can take an optional vox2ras transform, if one needs a specific one (default is affine transform from the given MRI data)
-  Can take `use_tkr`as input (optional), to be used in `apply_mri_transform`

jorgensd commented on src/scifem/biomedical.py:
```diff
@@ -16,7 +19,7 @@ def apply_mri_transform(
             https://www.mail-archive.com/freesurfer@nmr.mgh.harvard.edu/msg69541.html
             for more details. Else use the standard VOX2RAS transform (equivalent to attaching
-            an affine map to a nibabel image.
+            an affine map to a nibabel image.)
```
Here we need to add documentation  of `vox2ras_transform`.

jorgensd commented on src/scifem/biomedical.py:
```diff
@@ -28,21 +31,41 @@ def apply_mri_transform(
+        mgh_header = image.header
+    elif isinstance(image, nibabel.nifti1.Nifti1Image):
+        mgh = nibabel.MGHImage(image.dataobj, image.affine)
+        orientation = nibabel.aff2axcodes(mgh.affine)
```
Is the orientation needed here?

cdaversin replied:
Nope, I removed it :)

jorgensd commented on src/scifem/biomedical.py:
```diff
@@ -80,19 +103,28 @@ def read_mri_data_to_function(
         mri_data_path: Path to MRI data
         mesh: The mesh to attach the MRI data to.
         cells: Subset of cells to evaluate the MRI data at. If None, all cells are used.
+        vox2ras_transform: Optional transformation matrix to convert from voxel to ras coordinates.
```
Should we sanity check the size of this transform (4x4), or let nibabel handle it?

jorgensd commented:
A test that illustrates how to use it would be nice 👼🏻

finsberg commented:
> Looks good from my end. @finsberg do you mind having a look?

Looks good, but would be nice to have some tests that test the following
- User provided `vox2ras_transform` give correct result
- User provided `vox2ras_transform` with wrong shape raises ValueError
- Test for both formats (i.e `nibabel.freesurfer.mghformat.MGHImage` and `nibabel.nifti1.Nifti1Image`) if possible.

cdaversin commented:
I agree, I will implement some tests ASAP

cdaversin commented:
The tests are now updated with both formats (nifti and mgh), and with user provided vox2ras affine transform (nibabel already handles the case where vox2ras doesn't have the right shape).  All tests passed :partying_face: 

Question : I have added `use_tkr` as an input to `read_mri_data_to_function` since it was in `apply_mri_transform` but could not be set by users. The tests fail with `use_tkr = True` due to assertion `np.all(ijk_rounded >= 0)` which fails in this case. I guess this is because the data we build in the test has RAS orientation and not tkRAS. I don't think we need to specifically test the "tkr case" but we should still give the possibility to use this parameter ?

jorgensd commented:
> The tests are now updated with both formats (nifti and mgh), and with user provided vox2ras affine transform (nibabel already handles the case where vox2ras doesn't have the right shape). All tests passed 🥳
> 
> Question : I have added `use_tkr` as an input to `read_mri_data_to_function` since it was in `apply_mri_transform` but could not be set by users. The tests fail with `use_tkr = True` due to assertion `np.all(ijk_rounded >= 0)` which fails in this case. I guess this is because the data we build in the test has RAS orientation and not tkRAS. I don't think we need to specifically test the "tkr case" but we should still give the possibility to use this parameter ?

Yes, we can’t test it. But we should still give users the option to use it, I think!
